llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -1781,18 +1781,25 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
         assert(ICA.getArgTypes().size() >= 2 &&
                "Expected VPIntrinsic to have Mask and Vector Length args and "
                "types");
+
+        ArrayRef<const Value *> NewArgs = ArrayRef(ICA.getArgs());
+        if (!ICA.isTypeBasedOnly())
```

Comment:
Why do we need isTypeBasedOnly? If it is type-based only, what's the harm in dropping front/back the args?

Reply:
If it's typeBasedOnly then the args will be empty so I think dropping the front and back will trap

Reply:
Maybe guard it with !NewArgs.empty() for clarity?